ssas/service/public/api.go

```diff
@@ -771,3 +771,55 @@ func validateAndParseToken(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	render.JSON(w, r, response)
 }
+
+func validateJWT(w http.ResponseWriter, r *http.Request) {
+	trackingID := uuid.NewRandom().String()
+	event := ssas.Event{Op: "V2-Introspect", TrackingID: trackingID, Help: "Calling from admin.validateJWT()"}
+	ssas.OperationCalled(event)
+
+	defer r.Body.Close()
+
+	var reqBody map[string]string
+	if err := json.NewDecoder(r.Body).Decode(&reqBody); err != nil {
+		jsonError(w, http.StatusText(http.StatusBadRequest), fmt.Sprintf("Invalid request body: %s", err))
+		return
+	}
+
+	// Get the token string
+	tokenS := reqBody["token"]
+	if tokenS == "" {
+		jsonError(w, http.StatusText(http.StatusBadRequest), `Missing "token" field in body`)
+		return
+	}
+
+	// parse jwt claims
+	claims := jwt.MapClaims{}
+
+	if _, _, err := new(jwt.Parser).ParseUnverified(tokenS, claims); err != nil {
```

Comment:
shouldn't you be calling system.VerifyClientSignedToken?

Reply:
But now that I've gone back and re-read the original ticket I'm not 100% sure anymore. @dhgreene, do you know if we need to verify that the `system_id` exists in the DB at this endpoint, or is it sufficient to check that the required claims exist and give the user feedback on what's missing (if anything)?